### ISSUE TITLE
V20 U2 released

### ISSUE THREAD

**Issue by c0ldhands** [qmexnetworks/3cx-zammad-bridge#9]
Hello,

V20 Update 2 has now been released. Is it now possible to customize the integration for V20?
https://www.3cx.com/blog/releases/v20-update-2/

**Comment by qmex-mp** [comment]
Hi,
sadly the functionality we need is not included in the normal API, the 3CX WebUI gets the needed status and information via an undocumented binary websocket. The API we need is the following:
https://www.3cx.com/blog/releases/v20-call-control-api/

We need the web version of this API which is stated as "coming soon"

**Comment by jmdv-es** [comment]
Hi, I've just started searching how to integrate Zammad and 3CX and looks like API for linux is now available - comming soon is just stated for windows, am I right?

[https://www.3cx.com/blog/releases/v20-call-control-api/](https://www.3cx.com/blog/releases/v20-call-control-api/)

Thanks.

**Comment by qmex-mp** [comment]
Update 4 for 3CX V20 has finally everything we need, I think there will be a V20 compatible version in January 2025

**Comment by jmdv-es** [comment]
That’s great! Thank you very much.

**Comment by qmex-mp** [comment]
We released a Beta yesterday:

https://github.com/qmexnetworks/3cx-zammad-bridge/releases/tag/v0.7.0-beta1

**Comment by ZenDevMaster** [comment]
Access to the API in v.20 requires the ENT license, right?

**Comment by qmex-mp** [comment]
Hi,
sadly it looks like this is true, our dev system is Enterprise and on a Pro System, die options exists but "create" button is grayed us. Sadly there is no other option for us so far to implement this without the new Call Control API.

**Comment by ZenDevMaster** [comment]
Yes it seems like this might be true and that is pretty sad. The API save button is greyed for us too and we are not on the Enterprise License. 

Perhaps it is possible to use the Journaling Feature in the CRM Integration with an intermediate Postgre or MySQL database - https://www.3cx.com/docs/crm-integration/#h.o5j1mbrhb8tz. This kind of implementation would be a little hacky but might facilitate automatically generating tickets in Zammad
